File: Data Structures & Algorithms/remove-duplicates-from-sorted-array/submission-0.py
class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        k = 0
        last_unique_num = nums[k]
        
        for i in range(len(nums)):
            if last_unique_num != nums[i]:
                last_unique_num = nums[i]
                k+=1
                nums[k] = last_unique_num

        return k + 1

        

File: Data Structures & Algorithms/minimum-difference-between-highest-and-lowest-of-k-scores/submission-0.swift
class Solution {
    func minimumDifference(_ nums: [Int], _ k: Int) -> Int {
        let nums = nums.sorted()
        var _max: Int
        var _min: Int
        var diff: Int = Int.max
        var l = 0
        for r in 0..<nums.count {
            _max = nums[r]

            if (r - l + 1  == k) {
                _min = nums[l]
                diff = min(diff, (_max - _min))
                l += 1

            }

        }

        return diff

    }
}
